python/pyiceberg/io/pyarrow.py

```diff
@@ -379,3 +400,95 @@ def _(_: StringType) -> pa.DataType:
 def _(_: BinaryType) -> pa.DataType:
     # Variable length by default
     return pa.binary()
+
+
+class _ConvertToArrowExpression(BooleanExpressionVisitor[pc.Expression]):
+    def visit_true(self) -> pc.Expression:
+        return pc.scalar(True)
+
+    def visit_false(self) -> pc.Expression:
+        return pc.scalar(False)
+
+    def visit_not(self, child_result: pc.Expression) -> pc.Expression:
+        return ~child_result
+
+    def visit_and(self, left_result: pc.Expression, right_result: pc.Expression) -> pc.Expression:
+        return left_result & right_result
+
+    def visit_or(self, left_result: pc.Expression, right_result: pc.Expression) -> pc.Expression:
+        return left_result | right_result
+
+    def visit_unbound_predicate(self, predicate: UnboundPredicate[L]) -> pc.Expression:
+        raise ValueError("Please bind the expression first")
+
+    def visit_bound_predicate(self, predicate: BoundPredicate[Any]) -> pc.Expression:
+        return _iceberg_to_pyarrow_predicate(predicate)
+
+
+@singledispatch
+def _iceberg_to_pyarrow_predicate(expr: BoundPredicate[str]) -> pc.Expression:
+    raise ValueError(f"Unknown expression: {expr}")
+
+
+@_iceberg_to_pyarrow_predicate.register(BoundIsNull)
+def _(bound: BoundIsNull[str]) -> pc.Expression:
+    return pc.field(bound.term.ref().field.name).is_null(False)
+
+
+@_iceberg_to_pyarrow_predicate.register(BoundNotNull)
+def _(bound: BoundNotNull[str]) -> pc.Expression:
+    return pc.field(bound.term.ref().field.name).is_valid()
+
+
+@_iceberg_to_pyarrow_predicate.register(BoundIsNaN)
+def _(bound: BoundIsNaN[str]) -> pc.Expression:
+    return pc.field(bound.term.ref().field.name).is_null(True)
+
+
+@_iceberg_to_pyarrow_predicate.register(BoundNotNaN)
+def _(bound: BoundNotNaN[str]) -> pc.Expression:
+    return ~pc.field(bound.term.ref().field.name).is_null(True)
+
+
+@_iceberg_to_pyarrow_predicate.register(BoundEqualTo)
+def _(bound: BoundEqualTo[str]) -> pc.Expression:
+    return pc.field(bound.term.ref().field.name) == bound.literal.value
+
+
+@_iceberg_to_pyarrow_predicate.register(BoundNotEqualTo)
+def _(bound: BoundNotEqualTo[str]) -> pc.Expression:
+    return pc.field(bound.term.ref().field.name) != bound.literal.value
```

Comment:
Should this negate `==`? I don't see `!=` in the [expression docs](https://arrow.apache.org/docs/python/generated/pyarrow.dataset.Expression.html).

Reply:
Looks like this is correct from tests.